### ISSUE TITLE
Duet support

### ISSUE THREAD

**Issue by Patrik-svobodik** [gdombiak/OctoPod#164]
Hi, I love this app, but I switched from octoprint to duet board, and I dont want to use octoprint with the board because it has its own web interface, it has a really easy API and I think it would not be too hard to implement, but I do not understand your code, I want it especially because of OctoPod’s very good watchOS app and I know you are a very good developer

**Comment by gdombiak** [comment]
Hey Patrick,

Short answer: Read that link and I think that it would be easier to start a new app rather than making Duet work with OctoPod.

Longer answer: OctoPrint offers 2 communication channels which OctoPod relies on. The HTTP RESTful API (somewhat similar to the one of your link) and Websockets. Websockets are used for real time updates in an efficient way to mobile devices so that they do not have to poll for new data to keep the user interface up to date. Duet do not seem to support websockets. The HTTP RESTful API is used for sending commands and could be used for reading statuses though OctoPod relies mainly on websockets for that. The API of OctoPrint is way richer than the one of Duet and OctoPod relies on that. For instance, there is no API to set temperatures of bed or extruders. There is no API to get history of temps. No API to start a print job. No API to pause/cancel/restart a print job. No API for system commands. No API for custom commands. No API for moving X,Y,Z axis. No API for going home. Moreover, Due documentation says that they support a single API call at a time which OctoPod does not enforce/do.

In summary, I would recommend either using OctoPrint with your board or create a new app that follows the concepts/guidelines of Duet board. You are welcome to explore adapting OctoPod to work with Duet but I think that the level of effort will be bigger than starting a new app. This new app would be way simpler and smaller in codebase.

**Comment by Patrik-svobodik** [comment]
You are talking about missing API, that is because it is done via gcode that is sent via the gcode send API, but OK, thanks for info, I will try

**Comment by fake-name** [comment]
Any chance this could be revisited? The DWC2 interface has changed substantially since the issue was opened. 

[It seems like they're using websockets now](https://github.com/Duet3D/DuetWebControl/blob/master/src/store/machine/connector/RestConnector.js#L18).

Forking and creating a separate app would result in not being able to monitor both from the same place, which would be annoying. 

> For instance, there is no API to set temperatures of bed or extruders. There is no API to get history of temps. No API to start a print job. No API to pause/cancel/restart a print job. No API for system commands. No API for custom commands. No API for moving X,Y,Z axis. No API for going home.

What? It's always been able to do this. It's just via the gcode interface. After all, why reimplement what already exists.
